muse_payment_adaptor/api_interface/muse_api_client.py

```diff
@@ -0,0 +1,158 @@
+import json
+import logging
+from abc import ABC
+from typing import List, Tuple, Union
+
+from pydantic import BaseModel
+
+from core.datetimes.ad_datetime import datetime
+from invoice.models import Bill, BillItem
+from muse_payment_adaptor.api_interface.dataclasses import BulkyPaymentSubmissionRequest, \
+    BulkyPaymentSubmissionRequestMessage, BulkyPaymentMessageHeader, BulkyPaymentSummary, BulkyPaymentPayListElement
+from muse_payment_adaptor.api_interface.muse_http_client import MuseAdaptorHttpClient
+from muse_payment_adaptor.api_interface.signature import create_signature_b64_string
+from muse_payment_adaptor.apps import MusePaymentAdaptorConfig
+from muse_payment_adaptor.models import PaymentRequest, PaymentRequestDetails
+
+logger = logging.getLogger(__name__)
+_default_http_client = MuseAdaptorHttpClient(MusePaymentAdaptorConfig.muse_base_uri)
+
+
+# region ABC
+
+class MuseApiClient(ABC):
+    def __init__(self, api_http_client: MuseAdaptorHttpClient = _default_http_client):
+        self.api_http_client = api_http_client
+
+    @property
+    def _api_endpoint(self) -> str:
+        """
+        This property specifies the exact endpoint to be used by the client. The endpoint will be joined with root path
+        from the MuseAdaptorHttpClient.
+        """
+        raise NotImplementedError("`api_endpoint` not implemented")
```

Comment:
When you're using ABC there's no need to call not implemented manually, additionally decorating property with 
```Python
    @abstractmethod
```
Should be sufficient

Reply:
fixed